### PR TITLE
feat: add defineIntegration utility

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Adds a new `defineIntegration` helper
- Updates how Astro DB extend hooks types

## Testing

Tests should still pass, added a unit test

## Docs

https://github.com/withastro/docs/pull/8056

### CODE REVIEW

File: packages/astro/package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "./client/*": "./dist/runtime/client/*",
     "./components": "./components/index.ts",
     "./components/*": "./components/*",
+    "./integration": "./integrations/index.ts",
```
[the remark this answers is not in the document]
Comment (florian-lefebvre) [reply packages/astro/package.json:56]: Yes probably thx!

**Comment by matthewp** [comment]
Great job! I think I have a few opinions on the API shape but we can follow up on that. Reminder that to get this merged we'll need:

1. Promote your RFC to stage 2.
2. Have a stage 3 RFC PR.
3. Under an experimental flag (unless we decide experimental is not needed).

We can do (1) through an RFC meeting. Would you be available to come to a future community call to present the stage 1 and see if we have consensus to move? If you can't (I know it's a little later for you), we can set up a special RFC call for this.

**Comment by florian-lefebvre** [comment]
My bad I was too happy to avoid getting this stale so completely forgot about the RFC process, setting back to draft for now

**Comment by florian-lefebvre** [comment]
There's still no consensus on this for now, so I'm closing it until we want to revisit. RFC at https://github.com/withastro/roadmap/discussions/886

**Pull request closed without merging**